os/20211014-s390x-secure-execution.md

```diff
@@ -0,0 +1,190 @@
+and contains `/etc/crypttab`, `/etc/luks/root` and `/etc/luks/boot` files.
+
+The reason why `boot` has to be encrypted is that without encryption hacker is able to mount it and modify its contents, for example add new kargs to BLS config file(s), so during system's lifetime that modification could affect `sd-boot` image.
+
+## Firstboot
+
+During guest provisioning new hostkey(s) is/are fetched and used for generation of `sd-boot`. User should specify them within ignition config with pre-defined name template: `/etc/se-hostkeys/ibm-z-hostkey-N` where `N` is a key number, several hostkeys could be used:
+
+```yaml
+storage:
+  files:
+    -
+      path: /etc/se-hostkeys/ibm-z-hostkey-1
+      overwrite: true
+      contents:
+        source: https://url.com/key1
+```
+
+System would also set a new random LUKS keys for both `boot` and `root` filesystems, so each node has its own LUKS keys.
+
+## Lifetime
+
+Secure Execution is enabled by 2 tools provided by IBM's s390-utils package:
+- `genprotimg` - tool to generate `sd-boot`
+- `zipl` - tool to install new bootloader
+
+Despite `zipl` is well known and used by CoreOS, it couldn't be executed same way as before. When system runs in Secure Execution, `ostree` and `rdcore` automatically detect that and perform all required steps.
+That means, that user won't notice any difference, both mentioned components would automatically regenerate `sd-boot` image and
+install it with `zipl` whenever required.
+
+# Implementation steps
+
+## Step 1: CoreOS with existing patches to `libostree` could be configured to perform in Secure Execution mode.
+
+Customer's public hostkey(s) should be placed into `/etc/se-hostkeys/` using name templates `ibm-z-hostkey-N`. Secure Execution is enabled by hands following https://www.ibm.com/docs/en/linux-on-systems?topic=linux-introduction.
+
+## Step 2: New `qemu-secex.qcow2` build artifact for CoreOS with enabled Secure Execution.
+
+`coreos-xyz-qemu-secex.qcow2` image with LUKS encrypted `/boot` and `rootfs` is available for download. This is done by using new `buildextend-secex` `coreos-assembler` command, which:
+ - adds new `/se` partition for `sd-boot`
+ - encrypts `/boot` and `/` with random LUKS keys
+ - stores LUKS keys and `crypttab` config to `/etc`
+ - enables a rough equivalent of the effect of TPM-bound LUKS on z15, by appennding the LUKS keys and `crypttab` to the `initramfs`, which is encrypted as part of `sd-boot`. Original `initramfs` under `/boot` is non-modified
+ - generates `/se/sd-boot` encrypted by universal public hostkey (valid for all Z15 and newer machines)
+ - `zipl`es `/se/sd-boot`, so `qemu-secex.qcow2` image could only be run with Secure Execution
+
+Customer should add his public hostkeys(s) into `ignition` config.
+> **Note:** We cannot use LUKS `integrity` option, because it doesn's support resizing of partitions, so `rootfs` cannot be growned during `firstboot`.
```

Comment:
```suggestion
> **Note:** In this step we will not use the LUKS `integrity` option, because it doesn't support resizing of partitions, so `rootfs` cannot be grown during firstboot.
```